trie/trie.go

```diff
@@ -31,6 +31,28 @@ type Trie struct {
 	sha      hash.Hash // hash calc for trie
 }
 
+// CopyTrie returns a copy of the given trie.
+func CopyTrie(t *Trie) (*Trie, error) {
+	trie := &Trie{
+		db:       t.db,
+		dbprefix: t.dbprefix,
+		sha:      t.sha,
+	}
+
+	if t.root == nil {
+		trie.root = t.root
+	} else {
+		rootnode, err := trie.loadNode(t.root.Hash())
```

Comment:
I think it's no need to invoke loadNode here, because the root has been handled by loadNode when the copied trie was constructed. consider the following:
    `copyTrie := *t
     return &copyTrie`

    at the same time,  error check can be removed in the method as well as its callers such as stateDB.GetCopy()